.github/workflows/validate-documentation.yml

```diff
@@ -16,7 +16,13 @@ jobs:
 
     steps:
     - name: check out code
-      uses: actions/checkout@v3
+      uses: actions/checkout@v3.1.0
+
+    - name: verify links by lycheeverse
+      uses: lycheeverse/lychee-action@v1.5.1
```

Comment:
do you know if this is used widely? Any idea how opentelemetry.io folks deal with this ?

Reply:
- https://github.com/open-telemetry/opentelemetry-go/blob/ad45631b53faa74191fcee37c8f010e520af67e1/.github/workflows/links.yml#L19
- https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/b0d4457610231e5e3a8af32f8856a9d0b0fbdf8d/.github/workflows/validate-documentation.yml#L28
- also proposed here: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/11

in opentelemetry.io it looks like it checks hyperlinks in HTML using `htmltest`: https://github.com/open-telemetry/opentelemetry.io/blob/9fb34a099bd99fd37dc526cc3a6c09e7d037fd15/Makefile#L19

Some OTel repos also use https://github.com/tcort/markdown-link-check (AFAIK it is used in otel-spec, otel-demo and otel-collector).

Personally, I prefer https://github.com/lycheeverse/lychee as:

1. it can be used not only for markdown
1. you can run it "natively" (created in Rust so you can just grab a binary)
1. it is under a GitHub organization (not a user)